crates/bevy_app/src/app.rs

```diff
@@ -599,6 +603,74 @@ impl App {
         self
     }
 
+    /// Registers the given function into the [`AppFunctionRegistry`] resource using the given name.
+    ///
+    /// To avoid conflicts, it's recommended to use a unique name for the function.
+    /// This can be achieved by either using the function's [type name] or
+    /// by "namespacing" the function with a unique identifier,
+    /// such as the name of your crate.
+    ///
+    /// For example, to register a function, `add`, from a crate, `my_crate`,
+    /// you could use the name, `"my_crate::add"`.
+    ///
+    /// Only functions that implement [`IntoFunction`] may be registered via this method.
+    ///
+    /// See [`FunctionRegistry::register`] for more information.
+    ///
+    /// # Panics
+    ///
+    /// Panics if a function has already been registered with the given name.
```

Comment:
Should we panic here or just return the `Result`?

If we return the `Result`, then users will be forced to `unwrap` it themselves, but it gives them more options on handling the error (not that they can't just get the `AppFunctionRegistry` directly).

Reply:
Long term, I would prefer is this returned a result. But since none of the other methods on `App` do currently, I vote we panic for consistency. We can take a more general look at error handling during app setup later.